frontend/packages/topology/src/components/contextmenu/ContextSubMenuItem.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import { DropdownMenu, DropdownItem } from '@patternfly/react-core';
+// FIXME fully qualified due to the effect of long build times on storybook
+import Popper from '@console/shared/src/components/popper/Popper';
+import { AngleRightIcon } from '@patternfly/react-icons';
+
+import './ContextSubMenuItem.scss';
+
+type ContextSubMenuItemProps = {
+  label: React.ReactNode;
+  children: React.ReactNode[];
+};
+
+const ContextSubMenuItem: React.FC<ContextSubMenuItemProps> = ({ label, children, ...other }) => {
+  const nodeRef = React.useRef<HTMLButtonElement>(null);
+  const subMenuRef = React.useRef<HTMLDivElement>(null);
+  const [open, setOpen] = React.useState(false);
+  const referenceCb = React.useCallback(() => nodeRef.current || { x: 0, y: 0 }, []);
+
+  return (
+    <>
+      <DropdownItem
+        {...other}
+        className="topology-context-sub-menu"
+        component={
+          <button
+            ref={nodeRef}
+            type="button"
+            // prevent this DropdownItem from executing like a normal action item
+            onClick={(e) => e.stopPropagation()}
+            // mouse enter will open the sub menu
+            onMouseEnter={() => setOpen(true)}
+            onMouseLeave={(e) => {
+              // if the mouse leaves this item, close the sub menu only if the mouse did not enter the sub menu itself
+              if (!subMenuRef.current || !subMenuRef.current.contains(e.relatedTarget as Node)) {
+                setOpen(false);
+              }
+            }}
+            onKeyDown={(e) => {
+              // open the sub menu on enter or right arrow
+              if (e.keyCode === 39 || e.keyCode === 13) {
+                setOpen(true);
+                e.stopPropagation();
+              }
```

Comment:
should there be a keycode 37 check for closing the submenu?

Reply:
Left arrow is being captured in the scope of the sub menu itself. See code below on line 80 of this file.